applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -1423,6 +1413,14 @@ void TwoFluidNavierStokes<TwoFluidNavierStokesData<2, 3>>::ComputeGaussPointEnri
     // Stabilization parameters
     constexpr double stab_c1 = 4.0;
     constexpr double stab_c2 = 2.0;
+    //energy_check_cut elements
+
+    // Mass correction term
+        double volume_error_ratio = 0.0;
+    if (rData.IsCut()) {
+        const double volume_error = rData.IsAir() ? rData.VolumeError : -rData.VolumeError;
```

Comment:
but... if element is cut then it is never air right? Its still unclear to me how this works. You are adding mass to cut elements but how do you know that the added mass is liquid and not air?

Reply:
Because you error always refers to the water.

Reply:
@ddiezrod you're right. I missed your point.